SegmSubmodules/Models.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+from keras_contrib.layers import CRF
+from keras.layers import TimeDistributed
+from keras.layers import Dense, Dropout
+from keras.layers.wrappers import Bidirectional
+from keras.layers import Input, GRU
+from keras.models import Model
+from keras.models import load_model
+from keras.utils import to_categorical
+
+
+SEQ_LEN = 100  # constant
+
+
+def build_model(embed_dim=128):
+    """Builds default baseline model with CRF as the last layer and custom object corresponding to CRF
+
+    Returns:
+        keras compiled model"""
+    # define model with CRF
+    input = Input(shape=(None, embed_dim,))
+    model = Bidirectional(GRU(units=64, return_sequences=True,
+                                  recurrent_dropout=0.2))(input)  # variational biGRU
+    model = Dropout(0.2)(model)
+    model = TimeDistributed(Dense(16, activation="relu"))(model)  # a dense layer as suggested by neuralNer
+    crf_layer = CRF(2)  # CRF layer, 2 classes (song and not-song)
+    out = crf_layer(model)  # output
+
+    model = Model(input, out)
+    # CRF layer requires special loss and metrics
+    model.compile(optimizer="adam", loss=crf_layer.loss_function, metrics=[crf_layer.accuracy])
+
+    return model
+
+
+def load_from_cpt(path):
+    """Loads keras baseline model with custom objects from checkpoint
+
+    Args:
+        path: path to the model checkpoint
+
+    Returns:
+        loaded keras model"""
+    model = load_model(path, custom_objects=__create_custom_objects())
+    return model
+
+
+def predict_mask_long_sample(x_data, crf_model):
```

Comment:
You can simplify the name of the function via adding an assert for the length of the sample